### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by jonludlam**
Signed-off-by: Jon Ludlam <jonathan.ludlam@citrix.com>

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -52,13 +53,21 @@ Vagrant.configure("2") do |config|
 # Fix gnome-terminal
     sudo localectl set-locale LANG="en_GB.utf8"
 
+# Docker
+    sudo apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
```
Comment (gaborigloi) [inline Vagrantfile:57]: adv -> add

File: Vagrantfile [hunk 2]
```diff
@@ -26,10 +28,9 @@ Vagrant.configure("2") do |config|
     sudo apt-get install -y opam m4 libxen-dev git
     opam init -a --compiler=4.02.3 -y
     opam remote add xs-opam git://github.com/xapi-project/xs-opam
-    opam install -y merlin ocp-indent utop lwt_react depext camlp4
+    opam install -y merlin ocp-indent ocp-index ocp-browser utop lwt_react depext camlp4
     sudo opam depext -y xapi
-    opam pin add xenctrl 0.9.32
-    opam install ezxenstore
+    opam pin add lwt 2.7.1
```
Comment (Frezzle) [inline Vagrantfile:33]: Because lwt 3.0.0 broke our build.

File: Vagrantfile [hunk 3]
```diff
@@ -26,10 +28,9 @@ Vagrant.configure("2") do |config|
     sudo apt-get install -y opam m4 libxen-dev git
     opam init -a --compiler=4.02.3 -y
     opam remote add xs-opam git://github.com/xapi-project/xs-opam
-    opam install -y merlin ocp-indent utop lwt_react depext camlp4
+    opam install -y merlin ocp-indent ocp-index ocp-browser utop lwt_react depext camlp4
     sudo opam depext -y xapi
-    opam pin add xenctrl 0.9.32
-    opam install ezxenstore
```
Comment (Frezzle) [inline Vagrantfile:32]: Because ezxenstore is now a xapi dependency, so it doesn't need installing manually.

File: Vagrantfile [hunk 4]
```diff
@@ -26,10 +28,9 @@ Vagrant.configure("2") do |config|
     sudo apt-get install -y opam m4 libxen-dev git
     opam init -a --compiler=4.02.3 -y
     opam remote add xs-opam git://github.com/xapi-project/xs-opam
-    opam install -y merlin ocp-indent utop lwt_react depext camlp4
+    opam install -y merlin ocp-indent ocp-index ocp-browser utop lwt_react depext camlp4
     sudo opam depext -y xapi
-    opam pin add xenctrl 0.9.32
```
Comment (Frezzle) [inline Vagrantfile:31]: Because xenctrl dependency is now fixed in xs-opam.

File: Vagrantfile [hunk 5]
```diff
@@ -10,13 +10,15 @@ Vagrant.configure("2") do |config|
 
   config.ssh.forward_x11 = true
 
-  config.vm.network "public_network"
```
Comment (Frezzle) [inline Vagrantfile:13]: To fix a wifi issue.

**Review by Frezzle — APPROVED** [review]

File: Vagrantfile (hunk 1 above)
Comment (gaborigloi) [reply Vagrantfile:57]: Not a typo, my bad, `apt-key adv` is an actual command :open_mouth: :flushed:

File: Vagrantfile (hunk 2 above)
Comment (mseri) [reply Vagrantfile:33]: The latest update of `xs-opam` is constraining lwt in the opam files to avoid this problem (and the need for this fix).

Comment (Frezzle) [reply Vagrantfile:33]: I'm trying this PR without this pin, just to be sure (because it failed for me an hour ago, but now I'm doubting myself).

Comment (mseri) [reply Vagrantfile:33]: Differently from what I was hoping with https://github.com/xapi-project/xs-opam/issues/77, the pin is necessary. The presence of old version of our libraries makes the solver always find a configuration without `lwt` constraints, so if something is installing `lwt 3.0.0` (e.g. `lwt_react` installed above) the repository becomes unusable. The annoying alternative is to install the other packages after `xapi` dependencies, but it's too weak as a solution. I think it's better to use the pin as done here.

**Review by mseri — APPROVED** [review]

File: Vagrantfile [hunk 6]
```diff
@@ -52,13 +53,21 @@ Vagrant.configure("2") do |config|
...
+    sudo apt-add-repository 'deb https://apt.dockerproject.org/repo ubuntu-xenial main'
+    sudo apt-get update
+    sudo apt-get install -y docker-engine
+    sudo usermod -aG docker vagrant
+
 # Verify xapi can be built
     git clone git://github.com/xapi-project/xen-api
     eval `opam config env`
-    cd xen-api; ./configure; make; make test; cd ..
+    cd xen-api; ./configure; make; make test > test.log; cd ..
```
Comment (Frezzle) [inline Vagrantfile:66]: Won't this statement still succeed if the tests fail?

Comment (Frezzle) [reply Vagrantfile:66]: It's no reason to stop the merge for this, so am going to merge now, but it's worth thinking about later.

**Review by Frezzle — APPROVED** [review]
All looks good. Everything working in my tests. Merging.

**Pull request merged**